mesonbuild/backend/ninjabackend.py

```diff
@@ -2850,7 +2851,7 @@ def _generate_single_compile_target_args(self, target: build.BuildTarget, compil
             # flags will be added in reversed order.
             for d in reversed(i.get_incdirs()):
                 # Add source subdir first so that the build subdir overrides it
-                (compile_obj, includeargs) = self.generate_inc_dir(compiler, d, basedir, i.is_system)
+                (compile_obj, includeargs) = self.generate_inc_dir(compiler, d, basedir, i.is_system, i.is_native_cross)
```

Comment:
I'm concerned that doing this in the Ninja backend is going to break the vs backend, do we have tests for this that run in Visual Studio?

Reply:
This method is local to the Ninja backend, so it won't affect other backends.